### ISSUE TITLE
Support TreeView item label templating

### ISSUE THREAD

**Issue by brunnerh** [carbon-design-system/carbon-components-svelte#1660]
Currently the `TreeView` does not allow much customization in how the items are displayed.
It would be helpful if an optional slot is provided that is passed the item instance, e.g.

```svelte
<TreeView {children}>
  <svelte:fragment slot="item" let:item>
     <span class:important={item.important}>{item.text}</span>
  </svelte:fragment>
</TreeView>
```

**Comment by theetrain** [comment]
This sounds appropriate to me, especially given the proposed state for slots in #1621.

**Comment by metonym** [comment]
Released in [v0.82.0](https://github.com/carbon-design-system/carbon-components-svelte/releases/tag/v0.82.0).

[Docs.](https://svelte.carbondesignsystem.com/components/TreeView#slottable-node)

```svelte
<TreeView children="{children}" let:node>
  {node.id}
  {node.text}
  {node.expanded}
  {node.selected}
  {node.disabled}
  {node.leaf} // True if the node is a leaf (node does not have children)
</TreeView>
```

**Comment by brunnerh** [comment]
I think this is too restrictive, though.

In my example I access the source item as is, so additional properties can be used for styling or data display.

Maybe that should be split into two props, `node` and `item` where node contains any metadata and `item` is just the object as it has been passed in as part of `children` (without any modification to maintain the prototype). Ideally `children` would use a generic type parameter so type information can be propagated to the slot.
